src/Features/CSharp/Portable/RemoveUnreachableCode/CSharpRemoveUnreachableCodeDiagnosticAnalyzer.cs

```diff
@@ -69,7 +62,7 @@ private void AnalyzeBlock(SyntaxNodeAnalysisContext context)
             // binding diagnostics directly on the SourceMethodSymbol containing this block, and
             // so it can retrieve the diagnostics at practically no cost.
             var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
-            var diagnostics = semanticModel.GetDiagnostics(block.Span, cancellationToken);
+            var diagnostics = semanticModel.GetDiagnostics(cancellationToken: cancellationToken);
```

Comment:
@mavasani does this still walk through all trees in the compilation to find right one and filter out diagnostics? or is it O(1) hashmap.

certainly better than before, but feels like we should make this O(1) if it is not already.

Reply:
but in this PR though.

Reply:
This will only force complete symbols in this file, so should be independent of number of files in the compilation.